ssh/ssh_args.go

```diff
@@ -17,35 +18,54 @@ type SSHArgs struct {
 
 	ForceTTY bool
 
-	PrivKeyFile    boshsys.File
-	KnownHostsFile boshsys.File
+	PrivKeyFile         boshsys.File
+	KnownHostsFile      boshsys.File
+	CmdExistenceChecker cmdExistenceChecker
 
-	socks5Proxy *proxy.Socks5Proxy
+	Socks5Proxy *proxy.Socks5Proxy
 	dialer      proxy.DialFunc
 }
 
+type cmdExistenceChecker interface {
+	CommandExists(cmdName string) (exists bool)
+}
+
 func NewSSHArgs(connOpts ConnectionOpts, result boshdir.SSHResult, forceTTY bool, privKeyFile boshsys.File, knownHostsFile boshsys.File) SSHArgs {
+	cmdRunner := boshsys.NewExecCmdRunner(boshlog.NewLogger(boshlog.LevelNone))
 	socks5Proxy := proxy.NewSocks5Proxy(proxy.NewHostKey(), nil)
 	boshhttpDialer := boshhttp.SOCKS5DialFuncFromEnvironment(net.Dial, socks5Proxy)
 	dialer := func(net, addr string) (net.Conn, error) {
 		return boshhttpDialer(net, addr)
 	}
 
 	return SSHArgs{
-		ConnOpts:       connOpts,
-		Result:         result,
-		ForceTTY:       forceTTY,
-		PrivKeyFile:    privKeyFile,
-		KnownHostsFile: knownHostsFile,
-		socks5Proxy:    socks5Proxy,
-		dialer:         dialer,
+		ConnOpts:            connOpts,
+		Result:              result,
+		ForceTTY:            forceTTY,
+		PrivKeyFile:         privKeyFile,
+		CmdExistenceChecker: cmdRunner,
+		KnownHostsFile:      knownHostsFile,
+		Socks5Proxy:         socks5Proxy,
+		dialer:              dialer,
 	}
 }
 
 func (a SSHArgs) LoginForHost(host boshdir.Host) []string {
 	return []string{host.Host, "-l", host.Username}
 }
 
+func formProxyOpt(existenceChecker cmdExistenceChecker, proxyHostString string) string {
+	if existenceChecker.CommandExists("connect-proxy") {
+		return fmtAsProxyCommandOpt("connect-proxy -S %s %%h %%p", proxyHostString)
+	}
+	// TODO: detect nc -h contains \-x.*proxy address or similar?
```

Comment:
@cwlbraa We want to convert this TODO into a story instead of leaving it as a TODO in the code.  Can you explain what it means so we can generate some actionable and clear acceptance criteria for such a story? We're going to merge it anyway.  Thanks!